rules/naming/src/Rector/Class_/RenamePropertyToMatchTypeRector.php

```diff
@@ -123,69 +120,38 @@ private function refactorClassProperties(ClassLike $classLike): void
                 continue;
             }
 
-            /** @var string $oldName */
-            $oldName = $this->getName($property);
-
             $expectedName = $this->expectedNameResolver->resolveForPropertyIfNotYet($property);
             if ($expectedName === null) {
                 continue;
             }
 
-            if ($this->shouldSkipProperty($property, $oldName, $expectedName, $conflictingPropertyNames)) {
+            $currentName = $this->getName($property);
+            $propertyType = $this->getObjectType($property);
+            $propertyClassLike = $property->getAttribute(AttributeKey::CLASS_NODE);
```

Comment:
Looks here is a good place to use Factory, maybe some logic could be implemented in Value Object's getters.
I'd prefer to implement Factory when I'll apply PrepoertyRenamer and ValueObject for other rules as a separate PR.

Reply:
> I'd prefer to implement Factory when I'll apply PrepoertyRenamer and ValueObject for other rules as a separate PR.

:+1:

Reply:
We could add those factoriies to `ValueObjectFactory` namespace :+1: